core/trino-main/src/main/java/io/trino/operator/function/StreamingWorkProcessor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.function;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.operator.WorkProcessor;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+
+import java.util.List;
+import java.util.Optional;
+
+public class StreamingWorkProcessor
+{
+    private boolean processEmptyInput;
+    private final int[] channelsForSource;
+
+    public StreamingWorkProcessor(boolean processEmptyInput, List<Integer> requiredChannels)
+    {
+        this.processEmptyInput = processEmptyInput;
+        channelsForSource = requiredChannels.stream().mapToInt(i -> i).toArray();
+    }
+
+    public WorkProcessor.Transformation<Page, List<Optional<Page>>> toTableFunctionInput()
+    {
+        return input -> {
```

Comment:
Do you have tests for this code?

Reply:
excludeColumn function uses this code path